graphics/MoltenVK/Portfile

```
@@ -3,11 +3,12 @@
 PortSystem          1.0
 PortGroup           github 1.0
 
-github.setup        KhronosGroup MoltenVK 1.1.9 v
+github.setup        KhronosGroup MoltenVK 1.2.4 v
+github.tarball_from releases
 epoch               1
-revision            1
+revision            0
 
-set sdkversion      1.3.211.0
+platforms           {darwin any >= 15}
```

If I'm not mistaken it should be:
```suggestion
platforms           {macosx any} {macosx >= 15}
```
And remove the `platforms macosx` line 3 lines down which would otherwise override it.

---

_Sigh_ guess I wasn’t so careful when making that commit…

I believe the current usage would be correct as the package built via the build would be viable for 10.11 and later versions of macOS, only thing would be is the usage of darwin or macosx as the platform.

---

> only thing would be is the usage of darwin or macosx as the platform

Well the port previously indicated that it required macOS. Now you've changed it to indicate that it also works on pure Darwin, which seems unlikely given that it requires Metal which I assume is only in macOS and not in pure Darwin. So I suggest you change `darwin` back to `macosx`.

---

> I believe the current usage would be correct

I was just going based off the documentation @jmroot left in his commit message at https://github.com/macports/macports-base/commit/28666188b01ef181a93b36b7468807ae558c5d3b. Your syntax `{darwin any >= 15}` is not one of the examples Josh gave so I don't know if it works.

---

> > only thing would be is the usage of darwin or macosx as the platform
> 
> Well the port previously indicated that it required macOS. Now you've changed it to indicate that it also works on pure Darwin, which seems unlikely given that it requires Metal which I assume is only in macOS and not in pure Darwin. So I suggest you change `darwin` back to `macosx`.

As far as I'm aware, Apple's graphics frameworks, such as Quartz and Metal, are closed source, and only exist as a part of macOS, but don't exist as a part of the open source Darwin OS. So I think that @ryandesign is correct on this one.

---

Changed back to macosx

---

> I was just going based off the documentation @jmroot left in his commit message at [macports/macports-base@2866618](https://github.com/macports/macports-base/commit/28666188b01ef181a93b36b7468807ae558c5d3b). Your syntax `{darwin any >= 15}` is not one of the examples Josh gave so I don't know if it works.

Support for this combination was added a bit later in 3d9856b. It both sets the platform segment of the archive filename to `darwin_any` so the same archive will be used by all darwin versions, and sets `known_fail yes` on darwin < 15.

This port goes a little bit outside the anticipated usage, since the same archive is not used by all darwin versions. But since the older darwin versions also get a different port version, I guess it works.

As Ryan knows, my preference is to never have a port change its version depending on context like the OS version, but instead have a different port (or subport) for each version we want to provide.

---

@jmroot so you’d prefer having say `moltenvk_legacy` or similar

I’d issues with the libsdl2 port that did this forcing another port for 10.6 support, as other portfiles do this already I’d assumed this was fine.

---

> @jmroot so you’d prefer having say `moltenvk_legacy` or similar

Or `MoltenVK-1.1.9`, or whatever.

> I’d issues with the libsdl2 port that did this forcing another port for 10.6 support, as other portfiles do this already I’d assumed this was fine.

Yes, dynamic versions are currently a fairly widespread practice, which is unfortunate IMO. Not everyone agrees.

---

@jmroot thinking about it having a versions subport seems more ideal as who knows when/if this will change again.

I’ll make MoltenVK into a stub and add a runtime dep of `MoltenVK-1.1.9` or `MoltenVK-latest` depending on macOS version that should be better and can always be expanded later.

---

Do you all think that MoltenVK is going to be one of those ports that needs to be split into version-suffixed subports? The Khronos Group has released Mac packages for MoltenVK that are able to be run on macOS all the way back to 10.11 El Capitan (which is the version of macOS where Apple introduced Metal) since some of MoltenVK's earliest releases. Unless the Khronos Group makes some kind of major update to Vulkan that would break backwards-compatibility, I can't really think of a reason why MoltenVK would need to be split into multiple subports. Or am I missing previous discussions about MoltenVK that I'm unaware of?

---

@jasonliu-- the VulkanSDK is packaged in the dumbest way possible with a dmg that can’t be mounted on older macOS versions and then a custom installer. The installed also changed.

They’ve refused to change how the VulkanSDK are packaged.

We can’t build from source using buildbots as modern Xcode versions lowest deployment target is 10.13.

The only viable solution is making use of the newly provided archives from GitHub releases.

---

> @jasonliu-- the VulkanSDK is packaged in the dumbest way possible with a dmg that can’t be mounted on older macOS versions and then a custom installer. The installed also changed.
>
> They’ve refused to change how the VulkanSDK are packaged.

Now that you mention it, I do remember that happening to certain DMGs after APFS came along, although I think most projects figured out how to create packages that avoided the issue. Sad to hear that the people at Khronos Group aren't doing it.

> We can’t build from source using buildbots as modern Xcode versions lowest deployment target is 10.13.

And I take it that building from source is failing on 10.12? I know that attempting to build from source failed for me on my 10.11 machine.

So are you implying that it's already impossible to get MoltenVK 1.2.4 to work on 10.11? Or maybe you are anticipating newer versions becoming impossible to work on 10.12 and 10.11 in future releases of MoltenVK? If that's true, then I would understand why you would want to split into version-numbered subports now, to be able to keep a "newest version that still worked on macOSes older than _x_" available into the future.